reactor-netty-core/src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -265,28 +269,39 @@ static ProxyProvider createFrom(Properties properties) {
 	static ProxyProvider createHttpProxyFrom(Properties properties) {
 		String hostProperty;
 		String portProperty;
+		String userProperty;
+		String passwordProperty;
 		String defaultPort;
 		if (properties.containsKey(HTTPS_PROXY_HOST)) {
 			hostProperty = HTTPS_PROXY_HOST;
 			portProperty = HTTPS_PROXY_PORT;
+			userProperty = HTTPS_PROXY_USER;
+			passwordProperty = HTTPS_PROXY_PASSWORD;
 			defaultPort = "443";
 		}
 		else {
 			hostProperty = HTTP_PROXY_HOST;
 			portProperty = HTTP_PROXY_PORT;
+			userProperty = HTTP_PROXY_USER;
+			passwordProperty = HTTP_PROXY_PASSWORD;
 			defaultPort = "80";
 		}
 
 		String hostname = Objects.requireNonNull(properties.getProperty(hostProperty), hostProperty);
 		int port = parsePort(properties.getProperty(portProperty, defaultPort), portProperty);
 
+		String username = properties.getProperty(userProperty);
+		String password = properties.getProperty(passwordProperty);
```

Comment:
I would check that the password is non-null IF AND ONLY IF the user property is set. If username non-null but password null, fail with an NPE with a message like `"Proxy username is set via " + userProperty + ", but no password is set".

Reply:
I refactored my code to be more in line with the SOCKS part